tools/pipelines/templates/include-publish-npm-package.yml

```diff
@@ -14,72 +14,136 @@ parameters:
   type: boolean
   default: false
 
+- name: pool
+  type: object
+  default: Lite
+
 stages:
 - stage: publish_npm_internal_test
   dependsOn: build
   displayName: Publish Internal Test Packages
   condition: and(succeeded(), eq(variables['testBuild'], true))
+  variables:
+  - name: environment
+    value: test-package-build-feed
   jobs:
-  - template: include-publish-npm-package-steps.yml
-    parameters:
-      namespace: ${{ parameters.namespace }}
-      feedName: https://pkgs.dev.azure.com/fluidframework/internal/_packaging/test/npm/registry/
-      environment: test-package-build-feed
-      official: false
-      artifactPath: scoped
-  - ${{ if eq(parameters.publishNonScopedPackages, true) }}:  
-    - template: include-publish-npm-package-steps.yml
-      parameters:
-        namespace: false
-        feedName: https://pkgs.dev.azure.com/fluidframework/internal/_packaging/test/npm/registry/
-        environment: test-package-build-feed
-        official: false
-        artifactPath: non-scoped
+  - deployment: publish_${{ replace(variables.environment, '-', '_') }}
+    displayName: Publish ${{ variables.environment }}
+    pool: ${{ parameters.pool }}
+    environment: ${{ variables.environment }}
+    workspace:
+      clean: all
+    variables:
+      version: $[ stageDependencies.build.build.outputs['SetVersion.version']]
+      isLatest: $[ stageDependencies.build.build.outputs['SetVersion.isLatest']]
+    strategy:
+      runOnce:
+          deploy:
+            steps:
+            - download: current
+              artifact: pack
+            - task: UseNode@1
+              displayName: Use Node 12.x
+              inputs:
+                version: 12.x
+            - template: include-publish-npm-package-steps.yml
+              parameters:
+                namespace: ${{ parameters.namespace }}
+                feedName: https://pkgs.dev.azure.com/fluidframework/internal/_packaging/test/npm/registry/
+                official: false
+                artifactPath: scoped
+            - ${{ if eq(parameters.publishNonScopedPackages, true) }}:  
+              - template: include-publish-npm-package-steps.yml
+                parameters:
+                  namespace: false
+                  feedName: https://pkgs.dev.azure.com/fluidframework/internal/_packaging/test/npm/registry/
+                  official: false
+                  artifactPath: non-scoped
```

Comment:
can this made into a template two so you don't have do duplicate it three times?

Reply:
Updated to use a separate include-publish-npm-package-deployment.yml template 😄